lib/utils/isPrimitive.ts

```diff
@@ -0,0 +1,19 @@
+import type { util } from 'zod/v4/core';
+
+export function isPrimitive(value: unknown): value is util.Primitive {
```

Comment:
dev: introduce a new `isPrimitive` utility to indicate whether a value is primitive